src/main/java/bio/terra/common/ErrorCollector.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.common;
+
+import bio.terra.service.dataset.exception.IngestFailureException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ErrorCollector {
+  private List<String> errors;
+  private int maxBadLoadFileLineErrorsReported;
+  private String exceptionMessage;
+
+  public ErrorCollector(int maxBadLoadFileLineErrorsReported, String exceptionMessage) {
+    this.errors = new ArrayList<>();
+    this.maxBadLoadFileLineErrorsReported = maxBadLoadFileLineErrorsReported;
```

Comment:
This class could be used in other cases too so this field could have a more general name like `maxErrorsReported` or `maxErrors`.